### PR TITLE
support both input and data in eth_call, eth_estimateGas, personal_sendTransaction

### PR DESCRIPTION
**Pull request by JukLee0ira**
# Proposed changes
A user reported a bug in [here](https://www.xdc.dev/steven_lin_1581b1cdfd98a6/wip-ethestimategas-rpc-method-on-apothem-with-erc20-transfer-always-return-execution-reverted-error-5fgl) when using the Go-Ethereum client to call JSON-RPC. The issue arises because the parameter field 'data' has been renamed to 'input' from version 1.13 onwards.

This PR adds support of "input" field in `eth_call`,`eth_estimateGas` . Field "data" will be still supported
### Test
When this PR is not merged:
- For RPC="http://127.0.0.1:8545", the changes from this PR are applied.
- For RPC="https://arpc.apothem.network/", the changes from this PR are not applied.
#### eth_call
- `data` field

Request:
```shell
RPC="http://127.0.0.1:8545"
curl -s -X POST -H "Content-Type: application/json" ${RPC} -d '{
  "jsonrpc": "2.0",
  "id": 8001,
  "method": "eth_call",
  "params": [
    {
      "to": "0x0000000000000000000000000000000000000088",
      "data": "0x0db02622"
    },
    "latest"]}' | jq
```

Response:
```json
{
  "jsonrpc": "2.0",
  "id": 8001,
  "result": "0x000000000000000000000000000000000000000000000000000000000000000b"
}
```
- `input` field

Request:
```shell
RPC="https://arpc.apothem.network/"
curl -s -X POST -H "Content-Type: application/json" ${RPC} -d '{
  "jsonrpc": "2.0",
  "id": 8001,
  "method": "eth_call",
  "params": [
    {
      "to": "0x0000000000000000000000000000000000000088",
      "input": "0x0db02622"
    },
    "latest"]}' | jq
```

Response:
```json
{
  "jsonrpc": "2.0",
  "id": 8001,
  "error": {
    "code": -32000,
    "message": "execution reverted"
  }
}
```
---
Request:
```shell
RPC="http://127.0.0.1:8545"
curl -s -X POST -H "Content-Type: application/json" ${RPC} -d '{
  "jsonrpc": "2.0",
  "id": 8001,
  "method": "eth_call",
  "params": [
    {
      "to": "0x0000000000000000000000000000000000000088",
      "input": "0x0db02622"
    },
    "latest"]}' | jq
```

Response:
```json
{
  "jsonrpc": "2.0",
  "id": 8001,
  "result": "0x000000000000000000000000000000000000000000000000000000000000000b"
}
```
- Requests that include both an `input` field and a `data` field, and these fields have the same value.

Request:
```shell
RPC="http://127.0.0.1:8545"
curl -s -X POST -H "Content-Type: application/json" ${RPC} -d '{
  "jsonrpc": "2.0",
  "id": 8001,
  "method": "eth_call",
  "params": [
    {
      "to": "0x0000000000000000000000000000000000000088",
      "input": "0x0db02622",
      "data": "0x0db02622"
    },
    "latest"]}' | jq
```

Response:
```json
{
  "jsonrpc": "2.0",
  "id": 8001,
  "result": "0x000000000000000000000000000000000000000000000000000000000000000b"
}
```
- Requests that include both an `input` field and a `data` field, but these fields have different values.

Request:
```shell
RPC="http://127.0.0.1:8545"
curl -s -X POST -H "Content-Type: application/json" ${RPC} -d '{
  "jsonrpc": "2.0",
  "id": 8001,
  "method": "eth_call",
  "params": [
    {
      "to": "0x0000000000000000000000000000000000000088",
      "input": "0x0db02622",
      "data": "0x12345678"
    },
    "latest"]}' | jq
```

Response:
```json
{
  "jsonrpc": "2.0",
  "id": 8001,
  "result": "0x000000000000000000000000000000000000000000000000000000000000000b"
}
```
#### eth_estimateGas
- `data` field

Request:
```shell
RPC="http://127.0.0.1:8545"
curl -s -X POST -H "Content-Type: application/json" ${RPC} -d '{
  "jsonrpc": "2.0",
  "id": 8001,
  "method": "eth_estimateGas",
  "params": [
    {
      "to": "0x0000000000000000000000000000000000000088",
      "data": "0x0db02622"
    },
    "latest"]}' | jq
```

Response:
```json
{
  "jsonrpc": "2.0",
  "id": 8001,
  "result": "0x574a"
}
```
- `input` field

Request:
```shell
RPC="https://arpc.apothem.network/"
curl -s -X POST -H "Content-Type: application/json" ${RPC} -d '{
  "jsonrpc": "2.0",
  "id": 8001,
  "method": "eth_estimateGas",
  "params": [
    {
      "to": "0x0000000000000000000000000000000000000088",
      "input": "0x0db02622"
    },
    "latest"]}' | jq
```
Response:
```json
{
  "jsonrpc": "2.0",
  "id": 8001,
  "error": {
    "code": -32000,
    "message": "execution reverted"
  }
}
```

---
Request:
```shell
RPC="http://127.0.0.1:8545"
curl -s -X POST -H "Content-Type: application/json" ${RPC} -d '{
  "jsonrpc": "2.0",
  "id": 8001,
  "method": "eth_estimateGas",
  "params": [
    {
      "to": "0x0000000000000000000000000000000000000088",
      "input": "0x0db02622"
    },
    "latest"]}' | jq
```

Response:
```json
{
  "jsonrpc": "2.0",
  "id": 8001,
  "result": "0x574a"
}
```
- Requests that include both an `input` field and a `data` field, and these fields have the same value.

Request:
```shell
RPC="http://127.0.0.1:8545"
curl -s -X POST -H "Content-Type: application/json" ${RPC} -d '{
  "jsonrpc": "2.0",
  "id": 8001,
  "method": "eth_estimateGas",
  "params": [
    {
      "to": "0x0000000000000000000000000000000000000088",
      "input": "0x0db02622",
      "data": "0x0db02622"
    },
    "latest"]}' | jq
```

Response:
```json
{
  "jsonrpc": "2.0",
  "id": 8001,
  "result": "0x574a"
}
```
- Requests that include both an `input` field and a `data` field, but these fields have different values.

Request:
```shell
RPC="http://127.0.0.1:8545"
curl -s -X POST -H "Content-Type: application/json" ${RPC} -d '{
  "jsonrpc": "2.0",
  "id": 8001,
  "method": "eth_estimateGas",
  "params": [
    {
      "to": "0x0000000000000000000000000000000000000088",
      "input": "0x0db02622",
      "data": "0x12345678"
    },
    "latest"]}' | jq
```

Response:
```json
{
  "jsonrpc": "2.0",
  "id": 8001,
  "result": "0x574a"
}
```
## Types of changes

What types of changes does your code introduce to XDC network?
_Put an `✅` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Regular KTLO or any of the maintaince work. e.g code style
- [ ] CICD Improvement
- [x] N/A

## Impacted Components
Which part of the codebase this PR will touch base on,

_Put an `✅` in the boxes that apply_

- [ ] Consensus
- [ ] Account
- [ ] Network
- [x] Geth
- [ ] Smart Contract
- [ ] External components
- [ ] Not sure (Please specify below)

## Checklist
_Put an `✅` in the boxes once you have confirmed below actions (or provide reasons on not doing so) that_

- [ ] This PR has sufficient test coverage (unit/integration test) OR I have provided reason in the PR description for not having test coverage
- [ ] Provide an end-to-end test plan in the PR description on how to manually test it on the devnet/testnet.
- [ ] Tested the backwards compatibility.
- [x] Tested with XDC nodes running this version co-exist with those running the previous version.
- [ ] Relevant documentation has been updated as part of this PR
- [ ] N/A

### CODE REVIEW

File: internal/ethapi/transaction_args.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+	var accessList types.AccessList
+	if args.AccessList != nil {
+		accessList = *args.AccessList
+	}
+
+	balanceTokenFee := big.NewInt(0).SetUint64(gas)
+	balanceTokenFee = balanceTokenFee.Mul(balanceTokenFee, gasPrice)
+
+	// Create new call message
+	msg := types.NewMessage(addr, args.To, 0, value, gas, gasPrice, data, accessList, false, balanceTokenFee, number)
```
Comment (JukLee0ira) [inline internal/ethapi/transaction_args.go:163]: different with https://github.com/XinFinOrg/XDPoSChain/pull/527/commits/f3e46732c1d6944e30a7bb5cb8dc069d60ff3e9e

File: internal/ethapi/transaction_args.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+		id := (*hexutil.Big)(b.ChainConfig().ChainId)
+		args.ChainID = id
+	}
+	return nil
+}
+
+// ToMessage converts TransactionArgs to the Message type used by the core evm
+func (args *TransactionArgs) ToMessage(b Backend, number *big.Int, globalGasCap uint64) types.Message {
+	// Set sender address or use zero address if none specified.
+	addr := args.from()
```
Comment (JukLee0ira) [inline internal/ethapi/transaction_args.go:128]: different with https://github.com/XinFinOrg/XDPoSChain/pull/527/commits/f3e46732c1d6944e30a7bb5cb8dc069d60ff3e9e

**Comment by gzliudan** [comment]
Please use code format for request and json format for response, such as:

<pre>
```shell
// request code
```
 
```json
// response text
```
</pre>

File: internal/ethapi/transaction_args.go (hunk 1 above)
Comment (JukLee0ira) [reply internal/ethapi/transaction_args.go]: Different parameter counts require implementing EIP-1559

**Review by gzliudan — CHANGES_REQUESTED** [review]

**Review by gzliudan — CHANGES_REQUESTED** [review]

**Review by gzliudan — APPROVED** [review]
LGTM

**Review by wjrjerome — APPROVED** [review]

**Review by gzliudan — APPROVED** [review]

**Review by liam-lai — APPROVED** [review]

**Pull request merged**